docker-entrypoint.sh

```diff
@@ -108,7 +108,7 @@ if expr "$1" : "apache" 1>/dev/null || [ "$1" = "php-fpm" ] || [ "${NEXTCLOUD_UP
         count=0
         limit=10
 
-        if [ -f "$lock" ]; then
+        if [ -f "$lock" ] && [ -n "${NEXTCLOUD_INIT_LOCK+x}" ]; then
```

Comment:
Woah, clever! I wouldn't mind a redundant comment here, maybe "check that `NEXTCLOUD_INIT_LOCK` is set and _not_ set to null". I've never seen this parameter expansion syntax before today and it was a bit tricky to look up what it does. I've been reading & writing Bash for 20+ years.

Note to self: https://stackoverflow.com/questions/19097745/what-does-plus-colon-mean-in-shell-script-expressions , and `Omitting the colon results in a test only for a parameter that is unset.` in the `bash(1)` manual page.

Reply:
Have you tried this? Did it work for you?

Reply:
Hrm, sorry, I don't have a test environment set up and I don't want to test it on my "production"/live instance with real data. I'll look into making a test environment. It may take me a while.

Reply:
I added some steps to easily reproduce locally

Reply:
Added some more code comment